src/library/scala/collection/StrictOptimizedIterableOps.scala

```diff
@@ -255,4 +255,35 @@ trait StrictOptimizedIterableOps[+A, +CC[_], +C]
     coll
   }
 
+  /** A collection containing the last `n` elements of this collection.
+    * $willForceEvaluation
+    */
+  override def takeRight(n: Int): C = {
+    val b = newSpecificBuilder
+    b.sizeHintBounded(n, toIterable)
```

Comment:
Does this hint take into account that `n` may be way larger than the number of things available?

Reply:
Yes, that's why it's bounded by `toIterable`